src/Server/HTTP/HTMLForm.cpp

```diff
@@ -241,6 +241,9 @@ HTMLForm::MultipartReadBuffer::MultipartReadBuffer(ReadBuffer & in_, const std::
 
 bool HTMLForm::MultipartReadBuffer::skipToNextBoundary()
 {
+    if (in.eof())
+        return false;
```

Comment:
Below this function throws an exception `No boundary line found` if `in.eof()`. Probably it's better to be consistent, maybe we should return false there too? And with this check the assertion `assert(working_buffer.empty() || eof());` in the next line starts looking unnecessary.

Reply:
Also in this line https://github.com/ClickHouse/ClickHouse/blob/7792eaade6a7af22543bb365c47649defe4acacb/src/Server/HTTP/HTMLForm.cpp#L187 I think we should check the result of `skipToNextBoundary()` because it's important.

Reply:
Ok, I've reworked skipToNextBoundary a little bit as you suggested.